app.py

```diff
@@ -1,2 +1,40 @@
-from flask import Flask, request
+from flask import Flask, request, jsonify
 import json
+from flask.views import MethodView
+
+
+app = Flask(__name__)
+with open("restaurants.json", "r") as read_file:
+    data = json.load(read_file)
+
+restaurants = data['restaurants']
+
+
+class RestaurantEndpoint(MethodView):
+    def get(self):
+        return jsonify({'restaurants': restaurants}), 200
+
+    def post(self):
+        content = request.json
+        if content is not None:
+            restaurants[str(len(restaurants))] = content['restaurant_info']
+        return jsonify({'restaurants': restaurants}), 201
+
+
+class RestaurantItemEndpoint(MethodView):
+    def get(self, restaurant_id):
+        return jsonify({'info': restaurants[str(restaurant_id)]}), 200
+
+    def put(self, restaurant_id):
+        content = request.json
+        if content is not None:
+            restaurants[str(restaurant_id)] = content['restaurant_name']
+        return jsonify({'restaurants': restaurants}), 200
```

Comment:
`RestaurantItemEndpoint` processes single `restaraunt` info, so it is better to respond with a single `restaurant` that was changed, not a list of all restaurants.

when you think about proper input-output, take a look at your URLs:
* `/api/restaurants` -  restaurants is plural, restaurants is some collection (list), so operations on `/api/restaurants` level affect all collection
* `/api/restaurants/<int:restaurant_id>`- this endpoint refers to a particular restaraunt, so it should get/update/delete only one item from collection

Reply:
fixed